x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationUtils.java

```diff
@@ -120,6 +121,7 @@ public static void switchUserBasedOnActionOriginAndExecute(
 
         switch (actionOrigin) {
             case SECURITY_ORIGIN:
+            case POST_WRITE_REFRESH_ORIGIN:
```

Comment:
Note that when I tried to use the XPackUser.INSTANCE user below, it still complained about not having privileges to do unpromotable refreshes to indices such as .security*. So I think we need the extra priviliges of the security user here.

Reply:
I think we should create a new internal user for this. We have a bad habit of just reusing existing internal users which then leads to increasing privileges.
Let me see if I can pull together a quick PR to add a user for this purpose - I'll tentatively call it `_storage`, but I'd love suggestions.

Reply:
Hi thank you @tvernum ! Sounds good to me. If you need any help, feel free to ask me. The new user would need privileges to refresh user and also system indices such as .security* ones. For this reason the security role here worked, rather than the xpack user.